org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -303,11 +359,11 @@ private Optional<Either<Command, CodeAction>> getGetterSetterAction(CodeActionPa
 					TextEdit edit = operation.createTextEdit(pm, accessors);
 					return convertToWorkspaceEdit(context.getCompilationUnit(), edit);
 				};
-				return getCodeActionFromProposal(params.getContext(), context.getCompilationUnit(), ActionMessages.GenerateGetterSetterAction_label, kind, getAccessorsProposal);
+				return getCodeActionFromProposal(params.getContext(), context.getCompilationUnit(), actionMessage, kind, getAccessorsProposal);
 			} else {
-				Command command = new Command(ActionMessages.GenerateGetterSetterAction_ellipsisLabel, COMMAND_ID_ACTION_GENERATEACCESSORSPROMPT, Collections.singletonList(params));
+				Command command = new Command(ellipsisActionMessage, commandId, Collections.singletonList(params));

```

Comment:
Just wondering : If instead of passing `params: CodeActionParams` you passed a `ResolveUnimplementedAccessorsParams`, would that help eliminate the 3 separate registered commands ( https://github.com/redhat-developer/vscode-java/pull/2450/files#diff-ca92d1afe836a483259dfc56e55ada3e33076f195fd4f2e7a0fd06be4bfa664f ) on the vscode-java side ? Is it worth doing this ? We pass the `kind` from client to server via `ResolveUnimplementedAccessorsParams` to avoid having 3 versions of `java/resolveUnimplementedAccessors` for each case.

If it's not worth it, this approach is also fine.

Reply:
I think it could make our implementation clearer and it's worth doing. I have done it via the new commit https://github.com/eclipse/eclipse.jdt.ls/pull/2086/commits/a2d21b45082d12338cfd84b702e3e0bf9eedcc38